src/index.js

```diff
@@ -200,3 +200,48 @@ Hooks.on('canvasInit', () => {
 });
 
 Hooks.on('renderChatMessage', (_, html) => Item5e.chatListeners(html));
+
+Hooks.on("createToken", (newNpc) => {
+  // Checks if its a NPC type of actor and if the game setting is set to true
+  if (newNpc.actor.type === "npc" && game.settings.get("a5e", "npcHealthRandomization")) {
+    // getting the NPC Hit Dice
+    let hitDice = newNpc.actor.system.attributes.hitDice;
+
+    // get the NPC HP information
+    let npcHp = newNpc.actor.system.attributes.hp;
+    
+    // Get Constitution Modifier
+    let conMod = newNpc.actor.system.abilities.con.mod;
+
+    // Set variable to track total number of hitDice for con mod multiplier later
+    let hitDiceTotalCount = 0;
+  
+    // Build Hit Dice formula for Roll
+    let hitDiceFormula = "";
+    for (const prop in hitDice){
+      if(hitDice[prop].total > 0){
+        if (hitDiceFormula !== "") {
+          hitDiceFormula += "+";
+        } 
+        // Add hit dice to formula
+        hitDiceFormula += hitDice[prop].total + prop;
+        // Increment Hit Dice Count total
+        hitDiceTotalCount++
+      }
+    }
+
+    if (hitDiceTotalCount > 0) {
+      console.log(hitDiceFormula);
+      // Roll the hitDiceFormula
+      let newHp = new Roll(hitDiceFormula).roll();
```

Comment:
As of v9 rolls need to be async `const newHP = await new Roll('1d6').roll({ async: true });`

Reply:
`{ async: true }` is the default now, right? If so we shouldn't have to provide that.